packages/@aws-cdk/aws-lambda/test/function-hash.test.ts

```diff
@@ -123,8 +123,114 @@ describe('function hash', () => {
+      stack1 = new Stack();
+      layer1 = new lambda.LayerVersion(stack1, 'MyLayer', {
+        code: lambda.Code.fromAsset(path.join(__dirname, 'layer-code')),
+        compatibleRuntimes: [lambda.Runtime.NODEJS_12_X],
+        license: 'Apache-2.0',
+        description: 'A layer to test the L2 construct',
+      });
+      layer2 = new lambda.LayerVersion(stack1, 'MyLayer2', {
+        code: lambda.Code.fromAsset(path.join(__dirname, 'layer-code')),
+        compatibleRuntimes: [lambda.Runtime.NODEJS_12_X],
+        license: 'Apache-2.0',
+        description: 'A layer to test the L2 construct',
+      });
+    });
+
+    test('same configuration yields the same hash', () => {
+      const stack2 = new Stack();
+      const fn1 = new lambda.Function(stack2, 'MyFunction', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromInline('foo'),
+        handler: 'index.handler',
+        layers: [layer1],
+      });
+
+      const stack3 = new Stack();
+      const fn2 = new lambda.Function(stack3, 'MyFunction', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromInline('foo'),
+        handler: 'index.handler',
+        layers: [layer1],
+      });
+
+      expect(calculateFunctionHash(fn1)).toEqual(calculateFunctionHash(fn2));
+      expect(calculateFunctionHash(fn1)).toEqual('028f8a4cb1c719f29e70b7b3c0f2a9d7');
+    });
+
+    test('different layers impacts hash', () => {
+      const stack2 = new Stack();
+      const fn1 = new lambda.Function(stack2, 'MyFunction', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromInline('foo'),
+        handler: 'index.handler',
+        layers: [layer1],
+      });
 
+      const stack3 = new Stack();
+      const fn2 = new lambda.Function(stack3, 'MyFunction', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromInline('foo'),
+        handler: 'index.handler',
+        layers: [layer2],
+      });
+
+      expect(calculateFunctionHash(fn1)).toEqual('028f8a4cb1c719f29e70b7b3c0f2a9d7');
+      expect(calculateFunctionHash(fn2)).toEqual('e74647bf81c4d532137545c8234726f3');
+    });
+
+    describe('impact of lambda layer order on hash', () => {
+      test('without feature flag, preserve old behavior to avoid unnecessary invalidation of templates', () => {
+        const stack2 = new Stack();
+        const fn1 = new lambda.Function(stack2, 'MyFunction', {
+          runtime: lambda.Runtime.NODEJS_12_X,
+          code: lambda.Code.fromInline('foo'),
+          handler: 'index.handler',
+          layers: [layer1, layer2],
+        });
+
+        const stack3 = new Stack();
+        const fn2 = new lambda.Function(stack3, 'MyFunction', {
+          runtime: lambda.Runtime.NODEJS_12_X,
+          code: lambda.Code.fromInline('foo'),
+          handler: 'index.handler',
+          layers: [layer2, layer1],
+        });
+
+        expect(calculateFunctionHash(fn1)).not.toEqual(calculateFunctionHash(fn2));
```

Comment:
This test should assert the exact function hash, and not that they are different.